code/game/objects/items/weapons/pneumaticCannon.dm

```diff
@@ -15,84 +20,95 @@
 	var/obj/item/tank/internals/tank = null //The gas tank that is drawn from to fire things
 	var/gasPerThrow = 3 //How much gas is drawn from a tank's pressure to fire
 	var/list/loadedItems = list() //The items loaded into the cannon that will be fired out
-	var/pressureSetting = 1 //How powerful the cannon is - higher pressure = more gas but more powerful throws
+	var/pressure_setting = LOW_PRESSURE //How powerful the cannon is - higher pressure = more gas but more powerful throws
 
 /obj/item/pneumatic_cannon/Destroy()
 	QDEL_NULL(tank)
 	QDEL_LIST_CONTENTS(loadedItems)
 	return ..()
 
+/obj/item/pneumatic_cannon/proc/pressure_setting_to_text(pressure_setting)
+	switch(pressure_setting)
+		if(LOW_PRESSURE)
+			return "low"
+		if(MID_PRESSURE)
+			return "medium"
+		if(HIGH_PRESSURE)
+			return "high"
+		else
+			CRASH("Invalid pressure setting: [pressure_setting]!")
+
 /obj/item/pneumatic_cannon/examine(mob/user)
 	. = ..()
 	if(!in_range(user, src))
 		. += "<span class='notice'>You'll need to get closer to see any more.</span>"
 	else
+		. += "<span class='notice'>Use a <b>wrench</b> to change the pressure level. Current output level is <b>[pressure_setting_to_text(pressure_setting)]</b>.</span>"
 		if(tank)
 			. += "<span class='notice'>[bicon(tank)] It has \the [tank] mounted onto it.</span>"
 		for(var/obj/item/I in loadedItems)
 			. += "<span class='info'>[bicon(I)] It has \the [I] loaded.</span>"
 
-/obj/item/pneumatic_cannon/attackby(obj/item/W, mob/user, params)
-	..()
-	if(istype(W, /obj/item/tank/internals/) && !tank)
-		if(istype(W, /obj/item/tank/internals/emergency_oxygen))
-			to_chat(user, "<span class='warning'>\The [W] is too small for \the [src].</span>")
+/obj/item/pneumatic_cannon/return_analyzable_air()
```

Comment:
I'm not sure we even have this base proc on /obj/item to begin with

Reply:
There is Gas Analyzer PR on the ready which refactors Gas scans and adds TGUI. If TGUI doesn't work, I will push the other changes

Reply:
Ah, gotcha. I missed that part in the PR desc.